atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -576,45 +578,48 @@ private ColumnSelection getColumnSelectionForRange(RangeRequest range) {
             protected TokenBackedBasicResultsPage<RowResult<Set<Long>>, byte[]> getFirstPage() {
-                return getTimestampsPage(tableRef, rangeRequest, timestamp);
+                return getTimestampsPage(tableRef, rangeRequest, timestamp, GET_RANGE_OF_TS_MAX_BATCH);
             }
 
             @Override
             protected TokenBackedBasicResultsPage<RowResult<Set<Long>>, byte[]> getNextPage(
                     TokenBackedBasicResultsPage<RowResult<Set<Long>>, byte[]> previous) {
                 byte[] newStartRow = previous.getTokenForNextPage();
                 RangeRequest newRange = rangeRequest.getBuilder().startRowInclusive(newStartRow).build();
-                return getTimestampsPage(tableRef, newRange, timestamp);
+                return getTimestampsPage(tableRef, newRange, timestamp, GET_RANGE_OF_TS_MAX_BATCH);
             }
+
         };
         return ClosableIterators.wrap(rows.iterator());
     }
 
     private TokenBackedBasicResultsPage<RowResult<Set<Long>>, byte[]> getTimestampsPage(
             TableReference tableRef,
             RangeRequest range,
-            long timestamp) {
+            long timestamp,
+            long batchSize) {
         Stopwatch watch = Stopwatch.createStarted();
         try {
-            return runRead(tableRef, table -> getTimestampsPageInternal(table, range, timestamp));
+            return runRead(tableRef, table -> getTimestampsPageInternal(table, range, timestamp, batchSize));
         } finally {
             log.debug("Call to KVS.getTimestampsPage on table {} took {} ms.",
                     tableRef, watch.elapsed(TimeUnit.MILLISECONDS));
         }
     }
 
-    @SuppressWarnings("deprecation")
-    private TokenBackedBasicResultsPage<RowResult<Set<Long>>, byte[]> getTimestampsPageInternal(DbReadTable table,
-                                                                                                RangeRequest range,
-                                                                                                long timestamp) {
-        Comparator<byte[]> comp = UnsignedBytes.lexicographicalComparator();
+    private TokenBackedBasicResultsPage<RowResult<Set<Long>>, byte[]> getTimestampsPageInternal(
+            DbReadTable table,
+            RangeRequest range,
+            long timestamp,
+            long batchSize) {
+        Comparator<byte[]> comp = range.isReverse()
+                ? UnsignedBytes.lexicographicalComparator().reversed() : UnsignedBytes.lexicographicalComparator();
         SortedSet<byte[]> rows = Sets.newTreeSet(comp);
         int maxRows = getMaxRowsFromBatchHint(range.getBatchHint());
 
-
         try (ClosableIterator<AgnosticLightResultRow> rangeResults = table.getRange(range, timestamp, maxRows)) {
             while (rows.size() < maxRows && rangeResults.hasNext()) {
-                byte[] rowName = rangeResults.next().getBytes("row_name");
+                byte[] rowName = rangeResults.next().getBlob("row_name");
```

Comment:
`getBytes` was deprecated, replaced with `getBlob` which looks like it returns empty arrays rather than null if the column doesn't exist.